### PR TITLE
Merge bensu's custom key fn feature in with master and update readme …

### PR DESCRIPTION
**Pull request by ldenman**
Added option for :key-fn in server side and return :file and :response data on client side.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,9 +39,25 @@ Please follow Amazon's [official documentation](http://docs.aws.amazon.com/Amazo
 (def access-key "your-aws-access-key")
 (def secret-key "your-aws-secret-key")
 
+(defn my-key-fn []
+  (fn [r]
+    (let [rand-str (str (apply *' (list (rand-int 2147483647)
+                                        (System/currentTimeMillis))))]
+      (str (subs rand-str 0 9) "/" (:file-name r)))))
```
Comment (martinklepsch) [inline README.md]: Why the nested `fn` here?

**Comment by martinklepsch** [comment]
I wonder if a utility to generate object keys could be moved to the cljs part of this library to keep the server side simpler and more plug and play.

**Comment by GetContented** [comment]
@martinklepsch ah, yes, I'm in favour of this. I think I understand where this is at, now. I'll work on somesuch today.

**Comment by ldenman** [comment]
closing this PR since further progress is being made in other PR's.

**Pull request closed without merging**